feat: Add new ptn modules `azd/ml-hub-dependencies`.

Menghua1 opened the pull request:
## Description
Fixes https://github.com/Azure/Azure-Verified-Modules/issues/1227
<!--
>Thank you for your contribution !
> Please include a summary of the change and which issue is fixed.
> Please also include the context.
> List any dependencies that are required for this change.

Fixes #123
Fixes #456
Closes #123
Closes #456
-->

## Pipeline Reference

<!-- Insert your Pipeline Status Badge below -->

| Pipeline |
| -------- |
|  [![avm.ptn.azd.ml-hub-dependencies](https://github.com/Menghua1/bicep-registry-modules/actions/workflows/avm.ptn.azd.ml-hub-dependencies.yml/badge.svg?branch=ml-hub-dependencies)](https://github.com/Menghua1/bicep-registry-modules/actions/workflows/avm.ptn.azd.ml-hub-dependencies.yml) |

## Type of Change

<!-- Use the checkboxes [x] on the options that are relevant. -->

- [ ] Update to CI Environment or utilities (Non-module affecting changes)
- [x] Azure Verified Module updates:
  - [ ] Bugfix containing backwards-compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
    - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [ ] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [ ] Update to documentation

## Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] I have run `Set-AVMModule` locally to generate the supporting module files.
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings

<!--  Please keep up to date with the contribution guide at https://aka.ms/avm/contribute/bicep -->

@jongio for notification.

ReneHezser commented:
@Menghua1 
We have added examples for Bicep parameter files to the Readme. This has been applied to all published modules but needs to be done for PRs as well. Can you please run the Set-AVMModule utility as detailed [here](https://azure.github.io/Azure-Verified-Modules/contributing/bicep/bicep-contribution-flow/generate-bicep-module-files/). It is required for the validation pipeline to succeed and the contribution to be published.

Please reach out if any support is needed.

Menghua1 commented:
> @Menghua1 We have added examples for Bicep parameter files to the Readme. This has been applied to all published modules but needs to be done for PRs as well. Can you please update your branch and run the Set-AVMModule utility as detailed [here](https://azure.github.io/Azure-Verified-Modules/contributing/bicep/bicep-contribution-flow/generate-bicep-module-files/). It is required for the validation pipeline to succeed and the contribution to be published.
> 
> Please reach out if any support is needed.

@ReneHezser Thank you for your guidance. I have updated branch and re-ran the Set-AVMModule utility.

AlexanderSehr commented on .github/CODEOWNERS:
```diff
@@ -15,6 +15,7 @@
 /avm/ptn/azd/apim-api/ @Azure/avm-ptn-azd-apimapi-module-owners-bicep @Azure/avm-module-reviewers-bicep
 /avm/ptn/azd/container-apps/ @Azure/avm-ptn-azd-containerapps-module-owners-bicep @Azure/avm-module-reviewers-bicep
 /avm/ptn/azd/insights-dashboard/ @Azure/avm-ptn-azd-insightsdashboard-module-owners-bicep @Azure/avm-module-reviewers-bicep
+/avm/ptn/azd/ml-hub-dependencies/ @Azure/avm-ptn-azd-mlhubdependencies-module-owners-bicep @Azure/avm-module-reviewers-bicep
```
Team is either not existing or is not in the correct hierachy

AlexanderSehr replied:
and fixed.

AlexanderSehr commented on avm/ptn/azd/ml-hub-dependencies/main.bicep:
```diff
@@ -0,0 +1,475 @@
+  }
+}
+
+module keyVault 'br/public:avm/res/key-vault/vault:0.7.1' = {
```
Note that all references to AVM modules inside your module should pass the `enableTelemetry` value through

AlexanderSehr replied:
This should not have ben resolved.
Please note @Menghua1, I'm talking about a passthru - not a condition of the entire deployment.

You'd always want to deployment, independent of the telemetry. What I'm asking for is adding `enableTelemetry: enableTelemetry` to to params block of each module reference. Please do that and remove the condition. And don't resolve the comments, but just reply that you've completed the ask 😉

AlexanderSehr commented on avm/ptn/azd/ml-hub-dependencies/main.bicep:
```diff
@@ -0,0 +1,475 @@
+
+module storageAccount 'br/public:avm/res/storage/storage-account:0.9.1' = {
+  name: '${uniqueString(deployment().name, location)}-storage'
+  params: {
```
Misses passthru of `enableTelemetry` parameter

AlexanderSehr replied:
How come this was resolved? The parameter is still missing?

AlexanderSehr commented on avm/ptn/azd/ml-hub-dependencies/main.bicep:
```diff
@@ -0,0 +1,475 @@
+
+module cognitiveServices 'br/public:avm/res/cognitive-services/account:0.7.0' = {
+  name: '${uniqueString(deployment().name, location)}-cognitive'
+  params: {
```
Misses passthru of `enableTelemetry` parameter

AlexanderSehr replied:
How come this was resolved? The parameter is still missing?

AlexanderSehr commented on avm/ptn/azd/ml-hub-dependencies/main.bicep:
```diff
@@ -0,0 +1,475 @@
+
+module logAnalytics 'br/public:avm/res/operational-insights/workspace:0.6.0' = if (!empty(logAnalyticsName)) {
+  name: '${uniqueString(deployment().name, location)}-loganalytics'
+  params: {
```
Misses passthru of `enableTelemetry` parameter

AlexanderSehr replied:
How come this was resolved? The parameter is still missing?

AlexanderSehr commented on avm/ptn/azd/ml-hub-dependencies/main.bicep:
```diff
@@ -0,0 +1,475 @@
+
+module applicationInsights 'br/public:avm/ptn/azd/insights-dashboard:0.1.0' = if (!empty(applicationInsightsName) && !empty(logAnalyticsName)) {
+  name: '${uniqueString(deployment().name, location)}-insights'
+  params: {
```
Misses passthru of `enableTelemetry` parameter

AlexanderSehr replied:
How come this was resolved? The parameter is still missing?

AlexanderSehr commented on avm/ptn/azd/ml-hub-dependencies/main.bicep:
```diff
@@ -0,0 +1,475 @@
+
+module containerRegistry 'br/public:avm/res/container-registry/registry:0.4.0' = if (!empty(containerRegistryName)) {
+  name: '${uniqueString(deployment().name, location)}-registry'
+  params: {
```
Misses passthru of `enableTelemetry` parameter

AlexanderSehr replied:
How come this was resolved? The parameter is still missing?

AlexanderSehr commented on avm/ptn/azd/ml-hub-dependencies/main.bicep:
```diff
@@ -0,0 +1,475 @@
+
+module searchService 'br/public:avm/res/search/search-service:0.6.0' = if (!empty(searchServiceName)) {
+  name: '${uniqueString(deployment().name, location)}-searchservice'
+  params: {
```
Misses passthru of `enableTelemetry` parameter

AlexanderSehr replied:
How come this was resolved? The parameter is still missing?

AlexanderSehr commented on avm/ptn/azd/ml-hub-dependencies/main.bicep:
```diff
@@ -0,0 +1,475 @@
+
+module keyVault 'br/public:avm/res/key-vault/vault:0.7.1' = if (enableTelemetry) {
+  name: '${uniqueString(deployment().name, location)}-keyvault'
+  params: {
```
Missing passthru of `enableTelemetry` parameter

Menghua1 commented:
@AlexanderSehr Could you please check the [latest commit](https://github.com/Azure/bicep-registry-modules/pull/3241/commits/57090c1d666acdd866db57688deca653d79d0c3c)? The issues mentioned above have been fixed. Thank you!

AlexanderSehr commented:
> @AlexanderSehr Could you please check the [latest commit](https://github.com/Azure/bicep-registry-modules/pull/3241/commits/57090c1d666acdd866db57688deca653d79d0c3c)? The issues mentioned above have been fixed. Thank you!

oh I did 😉 Please read through my comments - in one of them I pointed out why the 'fix' was unfortunately not the correct one. There should not be any additional conditions added to the module deployments but just a parameter added in the params block

Menghua1 commented:
@AlexanderSehr I apologize for the oversight. Thank you for your feedback! I have submitted a new commit to address the issues correctly. Thank you very much.
